conductor-client/src/main/kotlin/com/openlattice/postgres/external/ExternalDatabasePermissioner.kt

```diff
@@ -492,69 +481,81 @@ class ExternalDatabasePermissioner(
-                extDbManager.connectToOrg(orgId)
-        ).toCompletableFuture().get()
-    }
-
-    private fun removePermissionsOnColumn(userRole: String, permissionRoles: List<UUID>): List<String> {
-        return permissionRoles.map {
-            revokeRoleSql(it.toString(), setOf(userRole))
-        }
-    }
-
-    private fun grantPermissionsOnColumnsOnTableToRoleSql(
-            privileges: Set<PostgresPrivileges>,
-            columns: String,
-            schemaName: String,
+    private fun removePermissionsOnColumnsOfTableSql(
+            userRole: String,
+            schema: Schemas?,
             tableName: String,
-            roleName: String
-    ): String {
-        val privilegeString = privileges.joinToString { privilege ->
-            "$privilege ( $columns )"
+            columnName: String,
+            privileges: List<PostgresPrivileges>
+    ): List<String> {
+        return privileges.map {
+            applyPermissionOnColumnOfTableToRoleSql(
+                    userRole, 
+                    schema,
+                    tableName,
+                    columnName, 
+                    it, 
+                    PgPermAction.REVOKE
+            )
         }
-        return """
-            GRANT $privilegeString 
-            ON $schemaName.${ApiHelpers.dbQuote(tableName)}
-            TO ${ApiHelpers.dbQuote(roleName)}
-        """.trimIndent()
     }
 
-    private fun grantPermissionsOnColumnSql(
+    private fun grantPermissionsOnColumnsOfTableSql(
             userRole: String,
-            column: TableColumn,
-            permissionRoles: List<UUID>
+            schema: Schemas?,
+            tableName: String,
+            columnName: String,
+            privileges: List<PostgresPrivileges>
     ): List<String> {
         val sqls = mutableListOf<String>()
 
-        permissionRoles.forEach {
-            sqls.add(grantRoleToRole(it.toString(), setOf(userRole)))
+        privileges.forEach {
+            sqls.add(
+                applyPermissionOnColumnOfTableToRoleSql(
+                    userRole, 
+                    schema,
+                    tableName,
+                    columnName, 
+                    it, 
+                    PgPermAction.GRANT
+                )
+            )
 
-            if (column.schema != null) {
-                sqls.add(grantUsageOnSchemaSql(column.schema, userRole))
+            if (schema != null) {
+                sqls.add(grantUsageOnSchemaSql(schema, userRole))
             }
         }
 
         return sqls
     }
 
+    private fun applyPermissionOnColumnOfTableToRoleSql(
+            roleName: String, 
+            schema: Schemas?,
+            tableName: String,
+            columnName: String,
+            privilege: PostgresPrivileges,
+            action: PgPermAction
+    ): String {
+        val schemaName = if (schema != null) {
+            schema.toString() + "."
+        } else {
+            ""
+        }
+
+        return "DO\n" +
```

Comment:
Any reason not to use the fancy triple quote string?

```
return """
  DO $do$
  BEGIN
    IF ...
    ...
    END IF;
  END $do$;
""".trimIndent()
```

Reply:
Actually, I think it's ok to use the triple quote string template here. The only reason I did it like this is because it was done in a similar way before, and specifically only for this part of the code. I worry maybe there's something I'm missing, so I kept it the same way when I first did this. 

I am more confident now it should be fine though, so I will use the string template style here. What do you think?

Reply:
Yeah, I see now the reason it's done the way it's done (without raw string template): it's actually a little subtle due to how "$" is parsed within a string template. Nevertheless, this can be circumvented (from [this](https://kotlinlang.org/docs/basic-types.html#string-templates)):

> To insert the $ character in a raw string (which doesn't support backslash escaping) before any symbol, which is allowed as a beginning of an identifier, use the following syntax:
```
val price = """
${'$'}_9.99
"""
```